src/types.jl

```diff
@@ -252,3 +269,48 @@ function Base.show(io::IO, bp::BreakpointRef)
     end
     print(io, ')')
 end
+
+# Possible types for breakpoint condition
+const Condition = Union{Nothing,Expr,Tuple{Module,Expr}}
+
+abstract type AbstractBreakpoint end
+
+struct BreakpointSignature <: AbstractBreakpoint
+    f # Method or function
+    sig::Union{Nothing, Type}
+    line::Int # 0 is a sentinel for first statement
```

Comment:
How about making this a `lineoffset` from the first statement? That may facilitate more robust identity after revision.

Reply:
Could you give an example? If Revise has revised it, we create a new `FrameCode` which will have updated line numbers.

Reply:
This is conditioned on what I thought was your goal to try to preserve breakpoints through a revision. (You may be abandoning that goal, IIUC.) But suppose you wanted to preserve "break at the second line of the body" when you revise (1) a single `+` to a `-` somewhere in the body of your method, and (2) also make other changes elsewhere in the same source file that alter the absolute line numbers. Then if `BreakpointSignature` stored a line offset rather than an absolute line number, it would be unaffected by the fact that unrelated changes occurred elsewhere in the same file. Note that the user could specify the line numbers in absolute terms, I'm just proposing that when one creates the `BreakpointSignature` one should convert it to an offset.

Reply:
I don't know. Currently I feel like just keeping the line number is the most straight forward. FWIW, I don't see breakpoints on signatures at a given line to be used that much. Just adding file:line is likely more ergonomic.